code/modules/mob/living/carbon/human/human.dm

```diff
@@ -904,3 +904,85 @@
 	.["Make alien"] = "?_src_=vars;makealien=\ref[src]"
 	.["Make slime"] = "?_src_=vars;makeslime=\ref[src]"
 	.["Toggle Purrbation"] = "?_src_=vars;purrbation=\ref[src]"
+
+/mob/living/carbon/human/MouseDrop_T(mob/living/target, mob/living/user)
+	if((target != pulling) || (grab_state < GRAB_AGGRESSIVE) || (user != target))	//Get consent first :^)
+		. = ..()
+		return
+	buckle_mob(target, FALSE, TRUE, FALSE)
+	. = ..()
+
+/mob/living/carbon/human/buckle_mob(mob/living/M, force = FALSE, check_loc = TRUE, yes = FALSE)
+	if(!ishuman(M))
+		return
+	if(!yes)
+		return
+	if(!riding_datum)
+		riding_datum = new /datum/riding/human
+		riding_datum.ridden = src
+	if(buckled_mobs && ((M in buckled_mobs) || (buckled_mobs.len >= max_buckled_mobs)))
+		return
+	if(M.incapacitated(FALSE, TRUE) || incapacitated(FALSE, TRUE))
+		M.visible_message("<span class='boldwarning'>[M] can't hang onto [src]!</span>")
+		return
+	if(!equip_buckle_inhands(M))
+		M.visible_message("<span class='boldwarning'>[M] can't climb onto [src] because [M.p_their()] hands are full!</span>")
+		return
+	. = ..(M, force, check_loc)
+	riding_datum.handle_vehicle_offsets()
+
+/mob/living/carbon/human/unbuckle_mob(mob/living/M)
+	unequip_buckle_inhands(M)
+	. = ..(M)
+	riding_datum.restore_position(M)
+
+/mob/living/carbon/human/proc/unequip_buckle_inhands(mob/living/carbon/user)
+	for(var/obj/item/cyborgride_offhand/O in user.contents)
+		if(O.ridden != src)	//This shouldn't EVER happen.
+			continue
+		if(O.selfdeleting)
+			continue
+		else
+			qdel(O)
+	return TRUE
+
+/mob/living/carbon/human/proc/equip_buckle_inhands(mob/living/carbon/human/user)
+	var/obj/item/cyborgride_offhand/inhand1 = new /obj/item/cyborgride_offhand(user)
+	var/obj/item/cyborgride_offhand/inhand2 = new /obj/item/cyborgride_offhand(user)
+	. = 0
+	inhand1.rider = user
+	inhand1.ridden = src
+	inhand2.rider = user
+	inhand2.ridden = src
+	. += user.put_in_hands(inhand1, TRUE)
+	. += user.put_in_hands(inhand2, TRUE)
+	if(. < user.get_num_arms())
+		. = FALSE
+		unequip_buckle_inhands(user)
+
+/obj/item/cyborgride_offhand			//COPYPASTED CODE UNTIL CYBORG RIDING GETS IN SO I CAN REUSE.

```

Comment:
This kinda overlaps with cyborg riding but it would be neat if this thing's throw_at instead redirected to the rider